server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala

```diff
@@ -0,0 +1,82 @@
+package org.apache.james.jmap.rfc8621.contract
+
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.http.ContentType.JSON
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.apache.http.HttpStatus.SC_OK
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.rfc8621.contract.Fixture._
+import org.apache.james.utils.DataProbeImpl
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+trait VacationResponseSetMethodContract {
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString, BOB_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .build
+  }
+
+  @Test
+  def updateShouldSucceed(server: GuiceJamesServer): Unit = {
+    val request =
+      s"""
+         |{
+         |   "using": [ "urn:ietf:params:jmap:core",
+         |     "urn:ietf:params:jmap:mail",
+         |     "urn:ietf:params:jmap:vacationresponse" ],
+         |   "methodCalls": [
+         |       ["VacationResponse/set",
+         |           {
+         |                "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |                "update": {
+         |                    "singleton": {
+         |                      "id":"singleton",
```

Comment:
should not be there

Reply:
yes it is server-set, in fact you should add a test demonstrating that the update fails if you include this field.

Reply:
Actually no, server-set properties can be specified in an update, but they have to be the same value as on the server (so here "singleton" is valid)

https://jmap.io/spec-core.html#set =>
```
Any server-set properties MAY be included in the patch if their value is identical to the current server value (before applying the patches to the object). Otherwise, the update MUST be rejected with an invalidProperties SetError.
```

I just wanted to say that it was not necessary I think to mention it in a test. But it's not wrong technically (I was not clear^^)

Reply:
The id is server-set now :)